Add support for IBGateway v978

StefanoRaggi opened the pull request:
- Updated compile-time references to IBGateway `v978` jar files.
- Added handling of the new **"Exit Session Setting"** popup window:
this window requires the user to choose the time to log off/restart IBGateway, in this PR we are disabling the feature by setting the log off/restart time to `10 minutes` before the current time.
- Updated recognition of the main window title to be compatible with both `v974` and `v978`.
- Added minor logging improvements.

Closes #6 
Closes #10

StefanoRaggi committed:
Add support for IBGateway v978

Martin-Molinero commented on java/IBAutomater/src/ibautomater/WindowEventListener.java:
```diff
@@ -385,5 +395,61 @@ private boolean HandleFinancialAdvisorWarningWindow(Window window, int eventId)
+        if (title != null && title.contains("Exit Session Setting")) {
+            LocalDateTime time = java.time.LocalDateTime.now();
+
+            // disable logout/restart by setting logoff/restart time to the next day
```
nit - comment is off
Did we get `time.minusMinutes(10);` from testing or is it documented?

StefanoRaggi replied:
The comment is actually correct, perhaps not clear enough, this is what happens:
- at `3:07 AM` this window pops up:
![image](https://user-images.githubusercontent.com/7236689/78011609-17a6d000-7344-11ea-912e-b0e64ae05dfc.png)
- the `Cancel` button just closes the window, but will open again after a few seconds
- to prevent the logout, the only option is to move the time away from the current time: by setting the time to `2:57 AM` (10 minutes before the current time) we are basically telling IBGateway to not perform the logout until tomorrow at `2:57 AM`; a few minutes before that time the window will open again and we will repeat the process

The 10 minutes value is an arbitrary choice, any positive value should be ok.

Martin-Molinero replied:
Got it 👍

Martin-Molinero commented on java/IBAutomater/src/ibautomater/WindowEventListener.java:
```diff
@@ -385,5 +395,61 @@ private boolean HandleFinancialAdvisorWarningWindow(Window window, int eventId)
+            if (button != null) {
+                this.automater.logMessage("Click button: [" + buttonText + "]");
+                button.doClick();
+            }
```
Should these `if (button != null)` have an else case which throws an exception?

StefanoRaggi replied:
All exceptions thrown from here would only be logged at line 99 but no harm in adding an extra log line, will fix

Martin-Molinero commented on java/IBAutomater/src/ibautomater/Common.java:
```diff
@@ -71,6 +72,18 @@ public static JToggleButton getToggleButton(Container container, String text) {
+        for (Component component : buttons) {
+            JRadioButton button = (JRadioButton)component;
+            String buttonText = button.getText();
+            if (buttonText == null || !button.getText().equals(text)) continue;
```
nit - could directly use `buttonText` instead of getting it again `button.getText()`. Maybe could use case in-sensitive comparison, I think `equalsIgnoreCase()`?

StefanoRaggi replied:
Yeah, should be `!buttonText.equals(text)`, will fix in the other methods too. 👍

StefanoRaggi committed:
Address review

- case insensitive text comparisons
- additional exceptions/logging for controls not found
